upgrade.py

```diff
@@ -415,15 +446,50 @@ def prepareTarget(self, progress_callback, primary_disk, target_boot_mode, boot_
                     return
             raise RuntimeError("Old partition layout is unsupported, run prepare_host_upgrade plugin and try again")
 
-    doBackupArgs = ['primary-disk', 'backup-partnum', 'boot-partnum', 'storage-partnum', 'logs-partnum', 'primary-partnum']
+    def backupFilesToTemp(self, progress_callback, target_disk, boot_device):
+
+        progress_callback(10)
+
+        # copy the files across:
+        primary_fs = util.TempMount(self.source.root_device, 'primary-', options=['ro'], boot_device=boot_device)
+        backup_dir = "/tmp/fs_backup"
+
+        files_list = ['etc/passwd', 'etc/group']
+        def restore_file(src_base, f, d):
+            src = os.path.join(src_base, f)
+            if os.path.exists(src):
+                files_list.append(f)
+
+        try:
+            os.mkdir(backup_dir, 0o755)
+
+            restore_list = self.buildRestoreList(primary_fs.mount_point)
+
+            self.handleRestoreList(primary_fs.mount_point, restore_list, restore_file)
+            logger.log("File list is:\n\t%s" % '\n\t'.join(files_list))
+            if util.runCmd2("tar -C '%s' -c -T - | tar -C '%s' -x" % (primary_fs.mount_point, backup_dir), inputtext='\n'.join(files_list)) != 0:
```

Comment:
Why are `etc/passwd` and `etc/group` handled specially with a double invocation of `tar`?

Reply:
Why specially? Normally they are not restored, but they are used in the restore process, so we need to save them. The double invocation is to copy all files and directories. Maybe I didn't get your point.